### PR TITLE
Chapter 14

### PR DESCRIPTION
**Pull request by Moonforeva**

### CODE REVIEW

**Comment by Moonforeva** [comment]
Ready

File: app/controllers/relationships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+  
+  def create
+    user = User.find(params[:followed_id])
```
Comment (saobv-2223) [inline app/controllers/relationships_controller.rb:5]:
```suggestion
    user = User.find_by params[:followed_id]
```

File: app/controllers/relationships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    current_user.follow(user)
```
Comment (saobv-2223) [inline app/controllers/relationships_controller.rb:6]:
```suggestion
    current_user.follow user
```

File: app/controllers/relationships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    respond_to do |format|
+      format.html { redirect_to @user }
+      format.js
+    end
+  end
+
+  def destroy
+    user = Relationship.find(params[:id]).followed
+    current_user.unfollow(user)
```
Comment (saobv-2223) [inline app/controllers/relationships_controller.rb:15]: remove ()

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -57,6 +57,20 @@ def destroy
     redirect_to users_url
   end
 
+  def following
+    @title = "Following"
+    @user = User.find(params[:id])
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb:62]: using find_by instead of find

File: app/models/user.rb [hunk 5]
```diff
@@ -68,13 +76,28 @@ def password_reset_expired?
...
+    following << other_user
+  end
+
+  def unfollow other_user
+    following.delete(other_user)
+  end
+
+  def following? other_user
+    following.include?(other_user)
+  end
```
Comment (saobv-2223) [inline app/models/user.rb:99]: remove ()

File: app/views/users/_follow_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<% unless current_user?(@user) %>
+  <div id="follow_form">
+  <% if current_user.following?(@user) %>
+    <%= render 'unfollow' %>
+  <% else %>
+    <%= render 'follow' %>
```
Comment (saobv-2223) [inline app/views/users/_follow_form.html.erb:6]: using "

**Comment by Moonforeva** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**